dnf/base.py

```diff
@@ -1451,15 +1451,15 @@ def trans_install(query, remove_query, comps_pkg):
                             _msg_installed(pkg)
                     sltr = dnf.selector.Selector(self.sack)
                     sltr.set(provides="({} if {})".format(comps_pkg.name, comps_pkg.requires))
-                    self._goal.install(select=sltr, optional=True)
```

Comment:
This one I'm not *100%* sure about, actually. I *think* it'll still do the right thing regarding the conditional - i.e. it won't cause a fatal error if the 'depended-upon' package isn't selected for install - but it's possible I'm wrong, let me know if so.

Reply:
Here it will fail if no provides or impossible to install

Reply:
It will fail here even in case of unavailable provide.

Reply:
But it can be solved by query in advance.

Reply:
OK, I'll see if I can figure that out.

Reply:
sigh, this still needs figuring out, I think.